homeassistant/components/husqvarna_automower/__init__.py

```diff
@@ -73,3 +81,23 @@ async def async_setup_entry(hass: HomeAssistant, entry: AutomowerConfigEntry) ->
 async def async_unload_entry(hass: HomeAssistant, entry: AutomowerConfigEntry) -> bool:
     """Handle unload of an entry."""
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+def cleanup_removed_devices(
+    hass: HomeAssistant, config_entry: ConfigEntry, available_devices: list[str]
+) -> None:
+    """Cleanup entity and device registry from removed devices."""
+    entity_reg = er.async_get(hass)
+    for entity in er.async_entries_for_config_entry(entity_reg, config_entry.entry_id):
+        if entity.unique_id.split("_")[0] not in available_devices:
+            _LOGGER.debug("Removing obsolete entity entry %s", entity.entity_id)
+            entity_reg.async_remove(entity.entity_id)
```

Comment:
Why are we removing entity entries separately? Are there entities that are not connected to a device entry? Removing a device entry will remove its connected entity entries and entities automatically.